ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -175,7 +175,10 @@ def external_brains():
                 _send_response(EnvironmentCommand.EXTERNAL_BRAINS, external_brains())
             elif req.cmd == EnvironmentCommand.RESET:
                 for k, v in req.payload.items():
-                    env_parameters.set_float_parameter(k, v)
+                    if isinstance(v, float):
+                        env_parameters.set_float_parameter(k, v)
+                    elif isinstance(v, list):
```

Comment:
This is where sending an entirely different command (not reset) might be useful.